Movement quick form (ported from farmOS 1.x)

mstenta opened the pull request:
Screenshot:

![Screenshot 2023-05-11 at 15-11-48 Movement Drush Site-Install](https://github.com/farmOS/farmOS/assets/95381/ae55ad0a-7f05-48e8-9ccc-c2861239a162)

paul121 commented on modules/quick/movement/src/Plugin/QuickForm/Movement.php:
```diff
@@ -0,0 +1,304 @@
+      '#type' => 'hidden',
+      '#value' => $this->combinedAssetGeometries($this->loadEntityAutocompleteAssets($form_state->getValue('asset'))),
+      '#prefix' => '<div id="asset-geometry" data-drupal-selector="asset-geometry">',
+      '#suffix' => '</div>',
```
Minor thing, but I've started using a `container` element instead of prefix/suffix for wrapper IDs. I prefer how this code looks/works using form API but curious what you think:

```php
    $form['asset_geometry_wrapper'] = [
      '#type' => 'container',
      '#attributes' => [
        'id' => 'asset-geometry',
        'data-drupal-selector' => 'asset-geometry',
      ],
      'asset_geometry' => [
        '#type' => 'hidden',
        '#value' => $this->combinedAssetGeometries($this->loadEntityAutocompleteAssets($form_state->getValue('asset'))),
      ],
    ];
```

mstenta replied:
https://github.com/farmOS/farmOS/pull/677/commits/cc7e2638a5d6c6e8c8fbc4becec211a6007cc350

paul121 commented on modules/quick/movement/src/Plugin/QuickForm/Movement.php:
```diff
@@ -0,0 +1,304 @@
+    $form['location_geometry'] = [
+      '#type' => 'hidden',
+      '#value' => $this->combinedAssetGeometries($this->loadEntityAutocompleteAssets($form_state->getValue('location'))),
+      '#prefix' => '<div id="location-geometry" data-drupal-selector="location-geometry">',
```
Is `data-drupal-selector` important here? This just feels like a special data ID that drupal may actually use for other things. I doubt there is any conflict here, but maybe we could rename to something like `data-movement-geometry` to make it clear we aren't using a special drupal form API keyword?

mstenta replied:
https://github.com/farmOS/farmOS/pull/677/commits/842d0031fe98aaf8820f2deaa015b967abed8f61

paul121 commented on modules/quick/movement/src/Plugin/QuickForm/Movement.php:
```diff
@@ -0,0 +1,304 @@
+    $form['date'] = [
+      '#type' => 'datetime',
+      '#title' => $this->t('Date'),
+      '#default_value' => new DrupalDateTime('midnight'),
```
This should use the user's timezone for reasons described here: https://github.com/farmOS/farmOS/pull/665

mstenta replied:
https://github.com/farmOS/farmOS/pull/677/commits/f1d5a18c64833eaf17c95242c8946ccddc54ca17

mstenta commented:
Thanks for the review+feedback @paul121! I've addressed all your comments above with fixup commits. Please take another look when you have a chance. :-)
